tests/integration/test_writes.py

```diff
@@ -424,6 +424,30 @@ def test_data_files(spark: SparkSession, session_catalog: Catalog, arrow_table_w
     assert [row.deleted_data_files_count for row in rows] == [0, 0, 1, 0, 0]
 
 
+@pytest.mark.integration
+def test_write_multiple_data_files(spark: SparkSession, session_catalog: Catalog, arrow_table_with_null: pa.Table) -> None:
```

Comment:
The test name implies it will write multiple data files, but the test body only writes 1 data file per overwrite. Shall we add a third overwrite which writes multiple data files to the table?

Reply:
yep! originally had more tests, but ran into #482. #501 should resolve this issue